lib/matplotlib/colorbar.py

```diff
@@ -582,8 +582,10 @@ def _add_solids(self, X, Y, C):
                 and any(hatch is not None for hatch in mappable.hatches)):
             self._add_solids_patches(X, Y, C, mappable)
         else:
+            # Ignore alpha if the user provided an explicit array
+            alpha = None if isinstance(self.alpha, np.ndarray) else self.alpha
```

Comment:
Sure, I updated `set_alpha` to handle both cases in the same place. It still seems unfortunate to get the colorbar's alpha out of sync with the mappable's alpha. But, I think that will only happen in this case.

Looks like there is https://github.com/matplotlib/matplotlib/issues/7430 with more discussion around a similar topic.

Reply:
I mean if you are planning to push through syncing them, we could do something else, like store a pointer to the mappable and just check it at draw time - I don't see any reason for the user to have the ability to set the alpha of the colorbar manually so that it is out of sync with the mappable, but maybe some folks want partially transparent colorbars?  But if so that would have to apply to the axes, ticks etc as well....

Reply:
I have no idea why the colorbar has a `set_alpha()`, I assume this may have been for when the colorbar+alpha looked blocky and allowed someone to force fully opaque just so it looks nicer. I could look into deprecating/modifying that in a separate issue/PR. I think something basic like this should get in for the releases as the current error is not descriptive of what the problem actually is.